swap_meet/vendor.py

```diff
@@ -1,2 +1,178 @@
+        return category_items
+        # except ValueError:
+            # pass
+            # return category_items
+
+    def swap_items(self, other_vender, my_item, their_item):
+        """
+        Returns True if items have been swapped between two inventory lists of two objects. 
+
+        If the vendor's inventory or other vender's invenory is an empty list, returns False.
+        If either of the items to be swapped not in appropriate lists, returns False
+
+        Parameters
+        ----------
+        other_vender: Vender instance object
+            The instance of the class Vendor
+        my_item: component
+            Item instance with given cateogry to be removed from self.inventory and appended to other_vender
+        their_item: component
+            Item instance with given cateogry to be removed from other_vender.inventory and appended to self.inventory
+
+        Returns
+        -------
+        True/False
+        """
+        if my_item not in self.inventory or their_item not in other_vender.inventory:
+            return False
+        elif not self.inventory or not other_vender.inventory:
+            return False
+        self.inventory.remove(my_item)
+        other_vender.inventory.remove(their_item)
+
+        self.inventory.append(their_item)
+        other_vender.inventory.append(my_item)
+        return True
+
+    def swap_first_item(self, other_vender):
+        """
+        Swaps the first item in self.inventory and other_vendor.inventory
+
+        If the vendor's inventory or other vender's invenory is an empty list, returns False.
+
+        Parameters
+        ----------
+        other_vender: Vender instance object
+            The instance of the class Vendor
+
+        Returns
+        -------
+        True/False
+        """
+        if not self.inventory or not other_vender.inventory:
+            return False
+
+        self.inventory.append(other_vender.inventory.pop(0))
+        other_vender.inventory.append(self.inventory.pop(0))
+        return True
+
+    def get_best_by_category(self, best_category):
+        """
+        Gets the first item with the highest condition within the category given.
+
+        If there are no items within the given category, returns None.
+
+        Parameters
+        ----------
+        best_category: str
+            String of the wanted category
+
+        Returns
+        -------
+        None/best_item
+        """
+        available_items = self.get_by_category(best_category)
+        item_hash = {}
+        index = 0
+
+        if not available_items:
+            return None
+
+        while index < len(available_items):
+            for item in available_items:
+                item_hash[item.condition] = index
+                index += 1
+
+        best_item_condition = max(item_hash)
+        item_index = item_hash[best_item_condition]
+        best_item = available_items[item_index]
+        return best_item
```

Comment:
Interesting approach! I like how you used the condition quality as the key! Very cool!